retrieval/target.go

```diff
@@ -235,6 +239,58 @@ func (t *Target) Update(cfg *config.ScrapeConfig, baseLabels, metaLabels clientm
 	t.metricRelabelConfigs = cfg.MetricRelabelConfigs
 }
 
+func newHTTPClient(cfg *config.ScrapeConfig) *http.Client {
+	tlsConfig := &tls.Config{}
+
+	// If a CA cert is provided then let's read it in so we can validate the
+	// scrape target's certificate properly.
+	if len(cfg.CACert) > 0 {
+		caCertPool := x509.NewCertPool()
+		// Load CA cert.
+		caCert, err := ioutil.ReadFile(cfg.CACert)
+		if err != nil {
+			log.Errorf("Unable to use specified CA cert %s: %s", cfg.CACert, err)
```

Comment:
No just return basically. As we can reload configs at runtime we absolutely don't want to crash when a config changes in a bad way.

Thinking about that, we should verify the existence of the cert files via `promtool check-config` – not in this PR though.